packages/pages/src/components/analytics/Analytics.ts

```diff
@@ -0,0 +1,192 @@
+import { TemplateProps } from "../../common/src/template/types";
+import { getRuntime, isProduction } from "../../util";
+import { AnalyticsMethods } from "./interfaces";
+import {
+  ConversionDetails,
+  CookieManager,
+  EntityPage,
+  PagesAnalyticsService,
+  providePagesAnalytics,
+  StaticPage,
+  Visitor,
+} from "@yext/analytics";
+
+/**
+ * The Analytics class creates a stateful facade in front of the \@yext/analytics
+ * Library's pagesAnalyticsProvider class. It takes in some data from the
+ * template configuration and uses it to provide configuration to the
+ * pagesAnalyticsProvider.
+ *
+ * Additionally, it provides handlers for controlling user opt-in for compliance
+ * requirements as well as for debugging, enabling conversion tracking, saving
+ * user identity information, and creating named analytics scopes for
+ * easy tagging.
+ *
+ * @public
+ */
+export class Analytics implements AnalyticsMethods {
+  private _optedIn: boolean;
+  private _conversionTrackingEnabled = false;
+  private _scopes: string[] = [];
+  private _cookieManager: CookieManager | undefined;
+  // TODO: create event queue for things to fire when optIn is called if we are not optedIn
```

Comment:
I don't want to block on doing this, I'm also not sure if that's strictly the correct thing to do. Need to ask product. It would lead to the time stamps being wrong, I guess. I've already set it up to track a pageView properly on opt in, not sure if the rest is necessary or not.